redhat-developer/service-binding-operator
# [WIP] Update ServiceBindingRequest.Status.Conditions when service not found

**This PR is on hold. Based on discussion with Predrag, I will pick up this PR in next sprint.**


### Motivation

When `Service` is not available, update `status.conditions` as
```
  conditions:
  - lastHeartbeatTime: "2020-05-28T20:07:51Z"
    lastTransitionTime: "2020-05-28T20:07:26Z"
    message: service not found
    reason: ServiceNotFound
    status: "False"
    type: Ready
```

### Changes

[1] Adds a check for Service Not Found case.

### How to test

1. Follow the steps given in this [example](https://github.com/redhat-developer/service-binding-operator/tree/master/examples/nodejs_postgresql) to:
- install DB operator, Service Binding Operator
- Import an application
- Create a DB instance 

2. Create the following `ServiceBindingRequest`:

```
apiVersion: apps.openshift.io/v1alpha1
kind: ServiceBindingRequest
metadata:
  name: binding-request
  namespace: service-binding-demo
spec:
  applicationSelector:
        resourceRef: nodejs-rest-http-crud 
        group: apps
        version: v1
        resource: deployments
 backingServiceSelector:
    group: postgresql.baiju.dev
    version: v1alpha1
    kind: Database
    resourceRef: db-demo
```

3. Use the following command to check if the status has been updated:
`oc get ServiceBindingRequest binding-request -o yaml`

The output should have `status.conditions.type` as 
-  `BindingReady` with `status: "False"`

```
status:
  conditions:
  - lastHeartbeatTime: "2020-05-28T20:07:51Z"
    lastTransitionTime: "2020-05-28T20:07:26Z"
    message: service not found
    reason: ServiceNotFound
    status: "False"
    type: Ready
  secret: ""

```

### Note

This PR does not deal with the scenario where `Service` is deleted and service is re-created with the same name.
